pocs/camera/camera.py

```diff
@@ -634,47 +699,54 @@ def _process_fits(self, file_path, info):
         fits_utils.update_observation_headers(file_path, info)
         return file_path
 
-    def _create_subcomponent(self, subcomponent, sub_name, class_name, base_class):
+    def _create_subcomponent(self, subcomponent, class_name):
         """
         Creates a subcomponent as an attribute of the camera. Can do this from either an instance
         of the appropriate subcomponent class, or from a dictionary of keyword arguments for the
         subcomponent class' constructor.
 
         Args:
-            subcomponent (instance of class_name | dict): the subcomponent object, or the keyword
+            subcomponent (instance of sub_name | dict): the subcomponent object, or the keyword
                 arguments required to create it.
-            sub_name (str): name of the subcomponent, e.g. 'focuser'. Will be used as the attribute
-                name, and must also match the name corresponding POCS submodule for this
-                subcomponent, e.g. `pocs.focuser`
-            class_name (str): name of the subcomponent class, e.g. 'Focuser'
-            base_class (class): the base class for the subcomponent, e.g.
-                `pocs.focuser.AbtractFocuser`, used to check whether subcomponent is an instance.
+            class_name (str): name of the subcomponent class, e.g. 'Focuser'. Lower cased version
+                will be used as the attribute name, and must also match the name of the
+                corresponding POCS submodule for this subcomponent, e.g. `pocs.focuser`.
         """
         if subcomponent:
+            base_module_name = "pocs.{}.{}".format(class_name.casefold(), class_name.casefold())
+            try:
+                base_module = load_module(base_module_name)
+            except AttributeError as err:
```

Comment:
I think I was getting `AttributeError` here during testing, yes.  `base_module_name` here is something like `pocs.<submodule_class_name>.<submodule_class_name>` and I think it ends up an `AttributeError` because the import of `pocs` succeeds? We test that again to confirm, though.

Reply:
OK, tested this and you are right, `load_module` does indeed raise `error.NotFound`. Both `AttributeError` and `ImportError` get raised along the way (so appear in the TraceBack) but they get caught lower down the stack.